include/exadg/utilities/throughput_parameters.h

```diff
@@ -123,10 +124,11 @@ struct ThroughputParameters
   void
   print_results(MPI_Comm const & mpi_comm)
   {
-    print_throughput(wall_times, operator_type, mpi_comm);
+    print_throughput(wall_times, ExaDG::Utilities::enum_to_string(operator_type), mpi_comm);
```

Comment:
Could we instead have a `print_throughput()` overload which takes an enums as we have for printing parameters?

Reply:
What do you mean by overload? A templated `template<typename OperatorType> void print_throughput()` function?

The `print_throughput()` function does not need more than the string. This is why I prefered to do it like this.

Reply:
I worked really hard to hide away the boilerplate `enum_to_string()` conversions in a lot of places to clean up the code. I think this is a fairly small change and keeps the whole `enum2string` stuff at one place in `print_functions.h`.

What you are describing can not happen due to the `constexpr` in https://github.com/exadg/exadg/blob/master/include/exadg/utilities/print_functions.h#L83.

I think the following could make both of us happy :)
```
if constexpr (Utilities::is_enum<ParameterType>())
   ...
else if consexpr (std::is_same_v<ParameterType, std::string>)
   ...
else
   static_assert(false,"print_throughput() can only be called with enum or string as second argument.")
```

Reply:
We should only introduce a template if we do not know the concrete type. For `print_throughput()` we know the type, namely `std::string`. `Utilities::enum_to_string()` is a utility function that one is free to use anywhere. The benefit of your hard work is still there, namely that this function is generic (and not written for every single enum type in ExaDG).

If you are strictly against using `Utilities::enum_to_string()` in the class `ThroughputParameters`, you would have to be also against `Patterns::Enum<OperatorType>()` a few lines above in my opinion (because in both cases we assume that the template parameter `OperatorType` is an enum. Note that there is concrete reasoning to use a template for `ThroughputParameters`, because we do not know the concrete enum definition (every PDE module in ExaDG defines its own enum for the operator type). This motivation is not given in my opinion for the function `print_throughput()`.